### PR TITLE
build(nexus): enhance nox session for building nexus

### PR DESCRIPTION
**Pull request by dmitryduev**
Fixes
-----
- Fixes WB-NNNNN
- Fixes #NNNN

Description
-----------
What does the PR do?

### <samp>🤖 Generated by Copilot at d073c07</samp>

No summary available (Limit exceeded: required to process 51028 tokens, but only 50000 are allowed per call)

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at a21b13d</samp>

> _Oh we are the `nexus` crew, we build for any platform_
> _We use the `nox` to automate, and `go` to cross-compile_
> _Heave away, me hearties, heave away_
> _We'll sail the seas with CUDA and `wget`, heave away_

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -7,27 +7,33 @@
...
-    """Builds the nexus binary for the current platform."""
-    system = platform.system().lower()
-    arch = "amd64" if platform.machine() == "x86_64" else platform.machine()
+    """Builds the nexus binary for the current platform.
+
+    Usage: nox -s build-nexus [target_system]
+
+    target_system: The target system to build for. Defaults to the current system.
+                   Use format: <system>-<arch> (e.g. linux-amd64, darwin-arm64)
+                   Use "all" to build for all known platforms.
```
Comment (kptkin) [inline noxfile.py:16]: Does all work?

File: nexus/scripts/build/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,4 @@
-ARG PYTHON_VERSION=3.9
-ARG PLATFORM=amd64
-FROM --platform=linux/${PLATFORM} python:${PYTHON_VERSION}
+FROM nvidia/cuda:12.2.0-runtime-ubuntu20.04
```
Comment (kptkin) [inline nexus/scripts/build/Dockerfile:1]: lol it actually worked

**Review by kptkin — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (dmitryduev) [reply noxfile.py:16]: yup, checked.

File: nexus/scripts/build/Dockerfile (hunk 2 above)
Comment (dmitryduev) [reply nexus/scripts/build/Dockerfile:1]: i know haha. gotta figure out what is missing/wrong in the original image

**Pull request merged**